internal/rpmmd/repository.go

```diff
@@ -287,3 +295,27 @@ func (pkg *PackageInfo) FillDependencies(rpmmd RPMMD, repos []RepoConfig, module
 	pkg.Dependencies, _, err = rpmmd.Depsolve([]string{pkg.Name}, nil, repos, modulePlatformID, false)
 	return
 }
+
+// FetchChecksum returns a repository checksum for given repo config
+func (rc *RepoConfig) FetchChecksum() (string, error) {
```

Comment:
Why isn't this a member function of `rpmmdImpl`? This way it's not mockable. I think we should rename `FetchPackageList` to something like `FetchRepositoriesMetadata` and not have this method. This way we can reduce the duplicate code and have mocking.

Reply:
Thanks for the suggestion. It is now implemented this way.